Addition: Name from heading

scottaohara opened the pull request:
closes #457

Related to the following:
- https://github.com/w3c/aria/pull/1860
- https://github.com/w3c/accname/pull/229 (this needs to be merged so the new links to 'accName: name from heading' will work

- [ ] TODO: add changelog entry when ready to merge


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/html-aam/pull/529.html" title="Last updated on Feb 8, 2024, 5:25 PM UTC (8131f4b)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/html-aam/529/e58672f...8131f4b.html" title="Last updated on Feb 8, 2024, 5:25 PM UTC (8131f4b)">Diff</a>

scottaohara committed:
Addition: Name from heading

closes #457

Related to the following:
- https://github.com/w3c/aria/pull/1860
- https://github.com/w3c/accname/pull/229 (this needs to be merged so the new links to 'accName: name from heading' will work

cookiecrook commented on index.html:
```diff
@@ -16451,6 +16471,26 @@ <h4>`summary` Element Accessible Name Computation</h4>
+          </li>
+          <li>
+            If the <a data-cite="accname-1.2/#dfn-accessible-name">accessible name</a> is still empty, then: if the `article` element has a 
+            <a href="https://dom.spec.whatwg.org/#concept-tree-descendant">descendant</a> that is a heading element (`h1`-`h6`), then use the subtree of the first such element
```
```suggestion
            <a href="https://dom.spec.whatwg.org/#concept-tree-descendant">descendant</a> that is a heading element (`h1`-`h6`), then use the accessible name of the first such element
```

scottaohara replied:
Can make this change. But doing this means either this PR or another PR would need to change all the other instances of referring to getting the name from the element’s subtree

cookiecrook replied:
Name isn’t always subtree, so I think it’d be good to keep this new addition accurate. 

I support changing the other instances, but I don’t think it has to be in the same PR. That said, if you’d prefer to keep it as is and change all instances separately, that’s fine too.

cookiecrook commented on index.html:
```diff
@@ -16435,6 +16435,26 @@ <h4>Other Form Elements Accessible Name Computation</h4>
+          </li>
+          <li>
+            If the <a data-cite="accname-1.2/#dfn-accessible-name">accessible name</a> is still empty, then: if the `dialog` element has a 
+            <a href="https://dom.spec.whatwg.org/#concept-tree-descendant">descendant</a> that is a heading element (`h1`-`h6`), then use the subtree of the first such element
```
```suggestion
            <a href="https://dom.spec.whatwg.org/#concept-tree-descendant">descendant</a> that is a heading element (`h1`-`h6`), then use the accessible name of the first such element
```

cookiecrook replied:
Should this mention that it's an iterative deepening depth-first search versus a breadth-first search, or is the cross reference to the AccName mention sufficient? I guess that's what the WPT tests will sort out, so this idea is optional of course.

scottaohara replied:
I linked to the accname step to specificity not repeat this (under the assumption that while this probably won’t change, you never really know)

cookiecrook reviewed (approved):
Approve with minor prose edit.

scottaohara committed:
Update index.html

Co-authored-by: James Craig <cookiecrook@users.noreply.github.com>

scottaohara committed:
Update index.html

Co-authored-by: James Craig <cookiecrook@users.noreply.github.com>

scottaohara committed:
Update index.html

per @cookiecrook review, changing a couple more instances where 'accessible name' should be used instead of 'subtree'

rahimabdi commented:
Closing, this PR has been migrated to the ARIA monorepo:
- https://github.com/w3c/aria/pull/2215
